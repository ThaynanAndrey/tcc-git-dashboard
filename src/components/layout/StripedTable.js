import React from 'react';

/**
 * Stateless component with the function of presenting a striped table.
 * 
 * @author Thaynan Nunes
 */
const StripedTable = ({ children }) => (
    <table className="striped highlight responsive-table">
        { children }
    </table>
);

export default StripedTable;