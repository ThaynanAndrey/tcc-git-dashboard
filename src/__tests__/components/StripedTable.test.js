import React from 'react';
import { shallow, configure } from "enzyme";
import toJson from 'enzyme-to-json';

import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

import StripedTable from '../../components/layout/StripedTable';

describe('Test StripedTable', () => {
  let tree;

  beforeEach(() => {
    tree = shallow(
      <StripedTable />
    );
  });

  it('renders correctly a table', () => {
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('the component should have class striped', () => {
    expect(tree.find(".striped").exists()).toBeTruthy();
  });

  it('the component should have class responsive-table', () => {
    expect(tree.find(".responsive-table").exists()).toBeTruthy();
  });
})