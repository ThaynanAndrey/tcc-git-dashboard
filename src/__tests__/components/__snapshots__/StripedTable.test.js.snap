// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test StripedTable renders correctly a table 1`] = `
<table
  className="striped highlight responsive-table"
/>
`;
